packages/shared/components/popups/MintNativeTokenFormPopup.svelte

```diff
@@ -0,0 +1,322 @@
+                        decimals,
+                        symbol,
+                        description,
+                        url,
+                        logoUrl,
+                    })
+                    await checkStronghold(mintAction, true)
+                }
+            } catch (err) {
+                if (!error) {
+                    error = err.error
+                        ? new BaseError({ message: err.error ?? err.message, logToConsole: true, saveToErrorLog: true })
+                        : err
+                }
+            }
+        }
+    }
+
+    async function validate(): Promise<boolean> {
+        try {
+            await Promise.all([
+                isNameValid(),
+                isTotalSupplyValid(),
+                isCirculatingSupplyValid(),
+                isDecimalsValid(),
+                isSymbolValid(),
+            ])
+            return true
+        } catch (error) {
+            console.error('Error: ', error)
+            return false
+        }
+    }
+
+    function isNameValid(): Promise<void> {
+        if (!name) {
+            nameError = 'Name is required'
+            return Promise.reject(nameError)
+        } else {
+            return Promise.resolve()
+        }
+    }
+
+    function isTotalSupplyValid(): Promise<void> {
+        if (totalSupply.toString().length < 1) {
+            totalSupplyError = 'Total supply is required'
+            return Promise.reject(totalSupplyError)
+        } else if (totalSupply < 1) {
+            totalSupplyError = 'Total supply must be greater than 0'
+            return Promise.reject(totalSupplyError)
+        } else {
+            return Promise.resolve()
+        }
+    }
+
+    function isCirculatingSupplyValid(): Promise<void> {
+        if (circulatingSupply.toString().length < 1) {
+            circulatingSupplyError = 'Circulating supply is required'
+            return Promise.reject(circulatingSupplyError)
+        } else if (circulatingSupply < 1) {
+            circulatingSupplyError = 'Circulating supply must be greater than 0'
+            return Promise.reject(circulatingSupplyError)
+        } else if (circulatingSupply > totalSupply) {
+            circulatingSupplyError = 'Circulating supply must be less than or equal to the total supply'
+            return Promise.reject(circulatingSupplyError)
+        } else {
+            return Promise.resolve()
+        }
+    }
+
+    function isDecimalsValid(): Promise<void> {
+        if (decimals.toString().length < 1) {
+            decimalsError = 'Decimals is required'
+            return Promise.reject(decimalsError)
+        } else if (decimals < 0) {
+            decimalsError = 'Decimals must be greater than or equal to 0'
+            return Promise.reject(decimalsError)
+        } else {
+            return Promise.resolve()
+        }
+    }
+
+    function isSymbolValid(): Promise<void> {
+        if (!symbol) {
+            symbolError = 'Symbol is required'
```

Comment:
Do we need some upper limit for letters?

Reply:
do we need that in general? i think we could use here again the json schema validator to simplify these validations, if it would work.

Reply:
There is no upper limit specified in the specification